src/react/providers/room-provider.tsx

```diff
@@ -0,0 +1,85 @@
+import { RoomOptions, RoomOptionsDefaults } from '@ably/chat';
+import React, { ReactNode, useEffect, useMemo } from 'react';
+
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatClient } from '../hooks/use-chat-client.js';
+
+/**
+ * Props for the RoomProvider component.
+ */
+export interface RoomProviderProps {
+  /** The id of the room. */
+  id: string;
+
+  /**
+   * The options to use when creating the room.
+   *
+   * @defaultValue {@link RoomOptionsDefaults}
+   */
+  options?: RoomOptions;
+
+  /**
+   * Set to false to disable auto-releasing the room when component unmounts,
+   * to support multiple RoomProviders for the same room.
+   *
+   * If set to false, you must manually release the room using
+   * `chatClient.rooms.release(id)` or have another RoomProvider for the same
```

Comment:
Is it worth just wrapping any chat type in a link? e.g. here {@link RoomProvider} ? :)